Core/Systems/Questing/Quests/MainPath/BlacksmithStartQuest.cs

```diff
@@ -0,0 +1,38 @@
+﻿using PathOfTerraria.Content.Items.Gear.Armor.Leggings;
+using PathOfTerraria.Content.NPCs.Town;
+using PathOfTerraria.Core.Systems.ModPlayers;
+using PathOfTerraria.Core.Systems.Questing.QuestStepTypes;
+using PathOfTerraria.Core.Systems.Questing.RewardTypes;
+using System.Collections.Generic;
+using Terraria.ID;
+
+namespace PathOfTerraria.Core.Systems.Questing.Quests.TestQuest;
+
+internal class BlacksmithStartQuest : Quest
+{
+	public override QuestTypes QuestType => QuestTypes.MainStoryQuestAct1;
+	public override string Name => "Forging a New Blade";
+
+	protected override List<QuestStep> _subQuests =>
+	[
+		new ParallelQuestStep([
+			new CollectCount(item => item.type == ItemID.IronOre || item.type == ItemID.LeadOre, 20, s => $"Collect {s} iron ore."),
+			new CollectCount(item => item.type == ItemID.IronHammer || item.type == ItemID.LeadHammer, 20, s => $"Collect {s} iron hammer."),
+		]),
+		new CollectCount(ItemID.StoneBlock, 50, s => $"Collect {s} stone."),
+		new CollectCount(ItemID.Wood, 20, s => $"Collect {s} wood."),
+	];
+
+	public override int NPCQuestGiver => ModContent.NPCType<Blacksmith>();
+
+	public override List<QuestReward> QuestRewards =>
+	[
+		new ActionRewards((p, v) =>
+			{
+				p.GetModPlayer<ExpModPlayer>().Exp += 500;
+				PoTItem.SpawnRandomItem(v);
+				PoTItem.SpawnItem<BurningRedBoots>(v);
+			},
+			"500 experience (POC giving experience)\nSome gear with an affix\nA unique item\nAgain, just for POC reasons"),
```

Comment:
This should be updated to have more in-line rewards for the state they would be at
Perhaps
* 500 Exp
* 1 Magic Random Sword (Item level 10 or below)
* 1 Magic Random Battleaxe (Item level 10 or below)

Reply:
I think I got this...but the quest book doesn't seem to do anything? I've applied this quest to the player, it shows up in the quest book, but clicking on it does nothing. The quest's tasks don't change, seemingly staying on the first test quest. I don't think I can test this as a result.

Reply:
Oh, specifically clearing the quest list then applying the quest works. That's really weird, but it's fine for testing.

Reply:
The message about left and right clicking to progress wasn't that it was done but rather need to figure that out. I'm not aware of any way to force progress a quest state forward and backward at this time.

Reply:
Couldn't you just spawn the items to progress the quest to completion through something like DL?

Reply:
The issue wasn't procurement, it was viewing the quest at all. With the test quest available, I was simply unable to switch to the "tab", so to speak, of the other quest. Now that I can view the right tab, the tasks overlap and seemingly don't update in any way, making it really difficult to see if you've progressed.
At the moment, by all means I should be done with the quest - nothing happened at all. I doubt it's on my end, though there's a nonzero chance it is. 

Apparently, restarting the quest and having every individual step done by the beginning worked, so maybe there's something between each step somehow? Unsure.